src/matrix/room/timeline/Timeline.js

```diff
@@ -211,8 +214,9 @@ export class Timeline {
     }
 
     /** @package */
-    replaceEntries(entries) {
+    async replaceEntries(entries) {
         this._addLocalRelationsToNewRemoteEntries(entries);
+        await this._loadRelatedEvents(entries);
```

Comment:
I'm slightly worried about this blocking getting the initial event add/update out, Especially in the worst case scenario where we hit the homeserver to get the context event ... Perhaps we could not await this call and send an update (e.g. call replaceEntries again) once we have obtained the context event.